config/routes.rb

```diff
@@ -4,10 +4,9 @@
   concern :range_searchable, BlacklightRangeLimit::Routes::RangeSearchable.new
   # For details on the DSL available within this file, see http://guides.rubyonrails.org/routing.html
 
-  class LoggedInConstraint
+  class CanSeeAdminConstraint
     def self.matches?(request)
-      current_user = request.env['warden'].user
-      !!current_user
+      AccessPolicy.new(request.env['warden'].user).can? :see_admin_pages
```

Comment:
OK, I guess since we aren't in a controller or temlate, you really do have to do this the "hard" way?  Good job figuring out the hard way!